### ISSUE TITLE
Fix `incrementals-publisher` chart issue: `artifactory.key is required`

### ISSUE THREAD

**Issue by lemeurherve** [jenkins-infra/helm-charts#656]
`helm template incrementals-publisher` output:

> Error: execution error at (incrementals-publisher/templates/secret.yaml:10:22): artifactory.key is required

Unit tests should be added, doc: https://github.com/helm-unittest/helm-unittest/blob/main/DOCUMENT.md

**Comment by SarathChandra24** [comment]
While running 
> helm template incrementals-publisher

we are getting the following 
Error: execution error at (incrementals-publisher/templates/secret.yaml:10:22): artifactory.key is required

This is because there aren't any default values specified in the values.yaml. I created the PR: https://github.com/jenkins-infra/helm-charts/pull/1077 to include default values.

Should the unit tests be written with custom_values_test.yaml in which we would specify the values and test it.

Can you please recommend on this.

**Comment by dduportal** [comment]
> While running
> 
> > helm template incrementals-publisher
> 
> we are getting the following Error: execution error at (incrementals-publisher/templates/secret.yaml:10:22): artifactory.key is required
> 
> This is because there aren't any default values specified in the values.yaml. I created the PR: #1077 to include default values.
> 
> Should the unit tests be written with custom_values_test.yaml in which we would specify the values and not editing the default values.yaml.
> 
> Can you please recommend on this.

The idea of the "default values" test is to mimick what would the command `helm template <...>` would do. It is currently using a custom values file while it should not.

I think you've got the right problem to solve, and the proposal is to solve it the same way it is solved on other charts such as `accoutapp`: you can see the PR here: https://github.com/jenkins-infra/helm-charts/pull/726/files

TL;DR; this PR allowed the `accountapp` chart to get rid of the "non default values" for the "default" test by removing the user error message about required values.

**Comment by lemeurherve** [comment]
Deployed with https://github.com/jenkins-infra/kubernetes-management/pull/5016
